canonical-web-and-design/maas-ui
# Fix angular routing issues when using $location.

## Done
- Replaced all instances of routing by `$location.path()` with `window.history.pushState` to work with singleSPA.
- Drivebys:
  - updated ui/.env.test to use `/l` angular basename
  - fixed routes in DhcpTarget to use angular basename env variable

## QA
- Check that adding a device from /dashboard routes to the devices list
- Check that deleting a node from node details routes to node list (machine, device, controller, pod)
- Check that deleting a fabric, VLAN, space or subnet details routes to network list
- Check that deleting a domain from domain details routes to domain list
- Check that deleting a zone from zone details routes to zone list
- Go to /MAAS/r/settings/dhcp and add a new snippet
- Change the type from "Global" and check that the "Applies to" link takes you to the right place

## Fixes
Fixes #1151
